solidity/test/bridge/Bridge.test.ts

```diff
@@ -1423,6 +1448,715 @@ describe("Bridge", () => {
+                })
+
+                context("when wallet state is MovingFunds", () => {
+                  // TODO: Just assert it passes without revert without
+                  //       repeating checks from Active state scenario.
+                })
+
+                context(
+                  "when wallet state is neither Active nor MovingFunds",
+                  () => {
+                    it("should revert", async () => {
+                      // TODO: Implementation. Make sure we check each other
+                      //       state in a separate sub-context.
+                    })
+                  }
+                )
+              }
+            )
+
+            context(
+              "when the single input doesn't point to the wallet's main UTXO",
+              () => {
+                it("should revert", async () => {
+                  // TODO: Implementation.
+                })
+              }
+            )
+          })
+
+          context("when input count is other than one", () => {
+            it("should revert", async () => {
+              // TODO: Implementation.
+            })
+          })
+        })
+
+        context("when main UTXO data are invalid", () => {
+          it("should revert", async () => {
+            // TODO: Implementation.
+          })
+        })
+      })
+
+      context("when there is no main UTXO for the given wallet", () => {
+        it("should revert", async () => {
+          // TODO: Implementation.
+        })
+      })
+    })
+
+    context("when transaction proof is not valid", () => {
+      context("when input vector is not valid", () => {
+        it("should revert", async () => {
+          // TODO: Implementation.
+        })
+      })
+
+      context("when output vector is not valid", () => {
+        it("should revert", async () => {
+          // TODO: Implementation.
+        })
+      })
+
+      context("when merkle proof is not valid", () => {
+        it("should revert", async () => {
+          // TODO: Implementation.
+        })
+      })
+
+      context("when proof difficulty is not current nor previous", () => {
+        it("should revert", async () => {
+          // TODO: Implementation.
+        })
+      })
+
+      context("when headers chain length is not valid", () => {
+        it("should revert", async () => {
+          // TODO: Implementation.
+        })
+      })
+
+      context("when headers chain is not valid", () => {
+        it("should revert", async () => {
+          // TODO: Implementation.
+        })
+      })
+
+      context("when the work in the header is insufficient", () => {
+        it("should revert", async () => {
+          // TODO: Implementation.
+        })
+      })
+
+      context(
+        "when accumulated difficulty in headers chain is insufficient",
+        () => {
+          it("should revert", async () => {
+            // TODO: Implementation.
+          })
+        }
+      )
+    })
+  })
```

Comment:
We could also add one testcase that goes through this line:
`delete mainUtxos[walletPubKeyHash];`

I think such testcase is not included at the moment.
I think this is an edge case that could only happen when `treasuryFee` would be 0.

Reply:
Nope, it is included within all cases which don't have a change output. This is `else` for `if outputsInfo.changeValue > 0)`. This has nothing TODO with `treasuryFee`. Unless I'm missing something?